### PR TITLE
[MI-2955]: Add prop to change input field to text area in autocomplete

### PR DESCRIPTION
**Pull request by aushwin**
- add prop to disable resize
- add prop to specify rows size
- add prop to remove close button
- made label optional prop

### CODE REVIEW

File: src/components/AutoComplete/AutoComplete.component.tsx [hunk 1]
```diff
@@ -189,21 +200,38 @@ export const AutoComplete = (props: AutoCompleteProps) => {
...
                 searchQuery={searchQuery}
-                onKeyDown={onKeyDown}
-                value={searchValue}
+                onKeyDown={(e) => {
+                    onKeyDownHandler(e);
+                    if (onKeyDown) {
+                        onKeyDown(e);
+                    }
+                }}
+                value={typeof value === 'undefined' ? searchValue : value}
```
Comment (sooraj-shukla) [inline src/components/AutoComplete/AutoComplete.component.tsx:217]:
```suggestion
                value={value ?? searchValue}
```

File: src/components/AutoComplete/AutoComplete.component.tsx [hunk 2]
```diff
@@ -157,10 +161,17 @@ export const AutoComplete = (props: AutoCompleteProps) => {
...
             const option = options[active];
-            setSearchValue(option.label ?? option.value);
+            if (!option) {
+                return;
+            }
+            if (typeof value !== 'undefined' && typeof onChange !== 'undefined' && open === true) {
+                onChange(option.label ?? option.value);
+            } else {
+                setSearchValue(option.label ?? option.value);
+            }
```
Comment (sooraj-shukla) [inline src/components/AutoComplete/AutoComplete.component.tsx:174]:
```suggestion
            onChange?.(option.label ?? option.value);
            setSearchValue(option.label ?? option.value);
```

File: src/components/AutoComplete/AutoComplete.component.tsx [hunk 3]
```diff
@@ -189,21 +200,38 @@ export const AutoComplete = (props: AutoCompleteProps) => {
...
                 fullWidth={fullWidth}
                 searchQuery={searchQuery}
-                onKeyDown={onKeyDown}
-                value={searchValue}
+                onKeyDown={(e) => {
+                    onKeyDownHandler(e);
+                    if (onKeyDown) {
+                        onKeyDown(e);
+                    }
+                }}
```
Comment (sooraj-shukla) [inline src/components/AutoComplete/AutoComplete.component.tsx:216]: Create a function

File: src/components/AutoComplete/AutoComplete.component.tsx [hunk 4]
```diff
@@ -189,21 +200,38 @@ export const AutoComplete = (props: AutoCompleteProps) => {
...
                 fullWidth={fullWidth}
                 searchQuery={searchQuery}
-                onKeyDown={onKeyDown}
-                value={searchValue}
+                onKeyDown={(e) => {
+                    onKeyDownHandler(e);
+                    if (onKeyDown) {
+                        onKeyDown(e);
+                    }
```
Comment (sooraj-shukla) [inline src/components/AutoComplete/AutoComplete.component.tsx:215]:
```suggestion
                  onKeyDown?.(e);
```

File: src/components/AutoComplete/AutoComplete.component.tsx [hunk 5]
```diff
@@ -189,21 +200,38 @@ export const AutoComplete = (props: AutoCompleteProps) => {
...
                 }}
                 onChange={(e) => {
                     setSearchQuery(e.target.value);
-                    setSearchValue(e.target.value);
+                    if (typeof value !== 'undefined' && typeof onChange !== 'undefined') {
+                        onChange(e.target.value);
+                    } else {
+                        setSearchValue(e.target.value);
+                    }
                 }}
```
Comment (sooraj-shukla) [inline src/components/AutoComplete/AutoComplete.component.tsx:235]: Create functions for all these.

File: src/components/AutoComplete/AutoComplete.component.tsx [hunk 6]
```diff
@@ -189,21 +200,38 @@ export const AutoComplete = (props: AutoCompleteProps) => {
...
+                        onKeyDown(e);
+                    }
+                }}
+                value={typeof value === 'undefined' ? searchValue : value}
                 label={label}
                 iconName={leadingIcon}
                 onClose={() => {
                     setOpen(false);
                     setSearchQuery('');
+                    if (typeof value !== 'undefined' && typeof onChange !== 'undefined') {
```
Comment (sooraj-shukla) [inline src/components/AutoComplete/AutoComplete.component.tsx:223]: Update it everywhere.

File: src/components/Input/Input.d.ts [hunk 7]
```diff
@@ -105,4 +106,27 @@ export interface InputProps {
...
+    component?: 'input' | 'textarea';
+
+    /**
+	 * If `true`, the resize option in the text area will be removed.
+	 * @default false
+	 */
+    disableResize?: boolean;
+
+    /**
+	 * The number of rows in the textarea.
```
Comment (sooraj-shukla) [inline src/components/Input/Input.d.ts:123]: Default?

**Review by sooraj-shukla — CHANGES_REQUESTED** [review]

**Review by sooraj-shukla — APPROVED** [review]

File: src/components/MMSearch/MMSearch.component.tsx [hunk 8]
```diff
@@ -186,10 +200,15 @@ export const MMSearch = (props: MMSearchProps) => {
...
             <Input
-                ref={inputRef}
+                ref={handleSetRef}
                 fullWidth={fullWidth}
                 searchQuery={searchQuery}
-                onKeyDown={onKeyDown}
+                onKeyDown={(e) => {
+                    onKeyDown(e);
+                    if (onKeyPress) {
+                        onKeyPress(e);
```
Comment (avas27JTG) [inline src/components/MMSearch/MMSearch.component.tsx:209]: any reason we need to do both?

Comment (aushwin) [reply src/components/MMSearch/MMSearch.component.tsx:209]: yes @avas27JTG , suppose the user wants to add any functionality on `onKeyPress` then he can pass it here. In our case (open ai) we need to override default behaviour of hitting enter for a text area (which goes to the next row) to on hitting enter should send the API request

Comment (avas27JTG) [reply src/components/MMSearch/MMSearch.component.tsx:209]: cool

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**